src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/WithThreadName.java

```diff
@@ -24,20 +24,46 @@
 
 package org.jenkinsci.plugins.workflow.support.concurrent;
 
+import java.lang.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.util.SystemProperties;
+
 /**
  * Utility to temporarily append some information to the name of the current thread.
  * This is helpful for making thread dumps more readable and informative:
  * stack trace elements do not contain any information about object identity.
  */
 public final class WithThreadName implements AutoCloseable {
-
+    /** Save original thread name to recover it in {@link #close} call.
+     * Remains {@code null} if activity of this class is {@link #disabled}
+     * on a particular deployment.
+     */
     private final String original;
 
+    /** Optional toggle via JVM properties to skip work here,
+     *  and forfeit easy debugging, e.g. on systems where
+     *  java.lang.Thread.setNativeName(Native Method) aka
+     *  JVM_SetNativeThreadName() and further platform
+     *  specific implementation takes inexplicably long.
+     */
+    private final static Boolean disabled = SystemProperties.getBoolean("DISABLE_WithThreadName");
```

Comment:
```suggestion
    private final static boolean enabled = SystemProperties.getBoolean(WithThreadName.class.getName() + ",enabled", true);
```
and then use `!enabled` elsewhere

Reply:
Fair enough :)

Reply:
Hm, is a comma okay in property name, or should that have been an underscore etc.?

With your proposal, would it go like `java -DWithThreadName,enabled=false ...` where I want it OFF?